### PR TITLE
Update metric check for resnet18 dynamo unit test

### PR DESCRIPTION
**Pull request by wonjoolee95**
Makes test compatible with https://github.com/pytorch/pytorch/pull/117434.

### CODE REVIEW

File: test/dynamo/test_dynamo.py
Comment (JackCaoG) [inline test/dynamo/test_dynamo.py:507]: do we know why we compile 2 more graphs?

Comment (mlazos) [reply test/dynamo/test_dynamo.py:507]: Yes because previously SGD was disabled, now it's enabled :)

**Review by mlazos — APPROVED** [review]

File: test/dynamo/test_dynamo.py
Comment (mlazos) [inline test/dynamo/test_dynamo.py:509]: Will this number also need to be changed?

Comment (wonjoolee95) [reply test/dynamo/test_dynamo.py:509]: Yep, it should -- I'm re-running the whole test suite will correct all the values.

File: test/dynamo/test_dynamo.py
Comment (JackCaoG) [reply test/dynamo/test_dynamo.py:507]: oh ok, that make sense!

**Comment by wonjoolee95** [comment]
@mlazos, this should be ready now. The process for merging these PRs should be:
1. Update the PyTorch PR (https://github.com/pytorch/pytorch/pull/117434) XLA pin to the commit pin of this PR -- `7d1e8e7597519eaae130a909f0b4d391ab1c5c32`.
2. Once PyTorch PR CI is green, merge it with the pin.
3. Then I'll rebase this PyTorch/XLA PR with PyTorch head and merge.

**Comment by mlazos** [comment]
> @mlazos, this should be ready now. The process for merging these PRs should be:
> 
> 1. Update the PyTorch PR ([Re-enable SGD pytorch#117434](https://github.com/pytorch/pytorch/pull/117434)) XLA pin to the commit pin of this PR -- `7d1e8e7597519eaae130a909f0b4d391ab1c5c32`.
> 2. Once PyTorch PR CI is green, merge it with the pin.
> 3. Then I'll rebase this PyTorch/XLA PR with PyTorch head and merge.

Hmm not sure if this needs to be rebased again, but I think another PR went in recently that also updated the XLA pin. https://github.com/pytorch/pytorch/commit/208e64a9ba2abca63ea77b3b2dc5a2f19e4c2ac4

**Comment by JackCaoG** [comment]
lol yea, we had similar issue on the other pr and we merged that one first. Can you rebase?

**Comment by JackCaoG** [comment]
Oh ok let me be more clear, I think we need to first
1. rebase this pr 
2. rebase pytorch pr to point to the new commit

This way your pytorch pr will point to a commit that can pass all CIs

**Comment by mlazos** [comment]
@wonjoolee95 can you rebase?

**Review by JackCaoG — APPROVED** [review]

**Comment by wonjoolee95** [comment]
> @wonjoolee95 can you rebase?

Yep, just rebased with master.

**Comment by mlazos** [comment]
@wonjoolee95 Can we rebase again lol, I fixed the issue that reverted the PR, and need to udpate the xla pin again

**Comment by wonjoolee95** [comment]
> @wonjoolee95 Can we rebase again lol, I fixed the issue that reverted the PR, and need to udpate the xla pin again

Sure thing! Just rebased -- also not that with each rebase, the commit pin will change. Now latest commit pin to pin is `2990cb38c17e06d0dbe25437674ca40130d76a8f`.

**Comment by mlazos** [comment]
https://github.com/pytorch/pytorch/pull/117434 landed finally, should be good to update the pin on this side

**Comment by wonjoolee95** [comment]
Merging this now to unblock head. Previously, CI was all green and locally the dynamo unit tests are passing.

**Pull request merged**
